spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -228,6 +228,16 @@ private DistributionMode adjustWriteDistributionMode(DistributionMode mode) {
     }
   }
 
+  private DistributionMode defaultDistributionMode() {
+    if (table.sortOrder().isSorted()) {
+      return RANGE;
+    } else if (table.spec().isPartitioned()) {
+      return HASH;
+    } else {
+      return NONE;
```

Comment:
if we keep this as hash, we will avoid small files even with unpartitioned tables by forcing the rebalance right?

Reply:
We have to request a valid `Distribution` in order for AQE to do its job. If the table is unpartitioned, I don't think we have what to cluster by.

Reply:
This is purely for inserts, though. I should probably call it `defaultInsertDistributionMode()` or something.

Reply:
I would think we could just cluster by all columns

Reply:
I am not sure how that would perform, to be honest. What about exploring this separately? Maybe, we can also handle this natively in Spark.